src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/nfs/NfsPermanentStorage.java

```diff
@@ -115,6 +131,23 @@ public boolean delete(SapsImage task) throws PermanentStorageException {
         return true;
     }
 
+    @Override
+    public List<AccessLink> generateAccessLinks(String taskId) {
+        List<AccessLink> taskDataLinks = new LinkedList<>();
+
+        String dirAccessLink = String.format(NFS_STORAGE_TASK_DATA_URL_PATTERN, this.baseUrl, taskId);
+
+        AccessLink inputDownloadingDirAccessLink = new AccessLink(INPUTDOWNLOADING_DIR, dirAccessLink + File.separator + INPUTDOWNLOADING_DIR);
+        AccessLink preprocessingDirAccessLink = new AccessLink(PREPROCESSING_DIR, dirAccessLink + File.separator + PREPROCESSING_DIR);
+        AccessLink processingDirAccessLink = new AccessLink(PROCESSING_DIR, dirAccessLink + File.separator + PROCESSING_DIR);
```

Comment:
Add a TODO comment to check for directories.

Reply:
Done